cmd/spire-server/cli/run/run.go

```diff
@@ -335,6 +339,31 @@ func newServerConfig(c *config) (*server.Config, error) {
 	sc.Telemetry = c.Telemetry
 	sc.HealthChecks = c.HealthChecks
 
+	// Write out deprecation warnings
+	switch {
+	case len(sc.PluginConfigs[upstreamca.Type]) == 0:
+		// no UpstreamCA configured
+	case c.Server.UpstreamBundle == nil:
+		// relying on the default upstream_bundle value of false
+		sc.Log.Warn(`Deprecation warning!
+The upstream_bundle configurable is undergoing deprecation (see issue #1095).
+A future release will remove the configurable and hard code the value to true.
+The configurable is currently set to the default value of false. Starting with
+the next release, the default value will be true. Consider migrating your
+deployment to an upstream_bundle value of true, ahead of the future
+deprecation. If that is not a possibility, you should at a minimum explicitly
+set the configurable to false to avoid a behavior change during upgrade.
+`)
+	case !sc.UpstreamBundle:
+		// explicitly setting upstream_bundle value to false
+		sc.Log.Warn(`Deprecation warning!
+The upstream_bundle configurable is undergoing deprecation (see issue #1095).
+A future release will remove the configurable and hard code the value to true.
+The configurable is currently set to false. Consider migrating your deployment
+to an upstream_bundle value of true, ahead of the future deprecation.
+`)
```

Comment:
Related to above comment, users will be safe for at least two full version cycles if they are explicitly setting it to false, so perhaps skip this warning for now? Once we get the default flipped, and we're sure we want to continue with full deprecation of the configurable, we can add this back?

Reply:
If we're not sure we're going to deprecate then I agree, warning them in advance now is probably not useful. I was under the impression we were for sure going to deprecate and wanted to give them as much time in advance to migrate, especially if we are planning on doing more frequent releases.

Reply:
Well, I would _like_ to deprecate it entirely, but it's possible there are users out there relying on this behavior for reasons we're not aware of... We know for sure the default should be true though, so to me it makes sense to take that step first.

I think it's ok to let this change marinate before making the next one... if we feel users need more time to prepare, we can deprecate over more than one release cycle?

Reply:
That's true. SGTM!